envoy/npidds.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+package pb.cilium;
+// package envoy.api.v2;
+
+import "api/discovery.proto";
+
+import "google/api/annotations.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Network policy identifier management and NPIDDS]
+
+// Each resource name is a host's IPv4 or IPv6 address, e.g. `10.1.2.3`.
+service NetworkPolicyIdentifierDiscoveryService {
+  rpc StreamNetworkPolicies(stream envoy.api.v2.DiscoveryRequest)
+      returns (stream envoy.api.v2.DiscoveryResponse) {
+  }
+
+  rpc FetchNetworkPolicies(envoy.api.v2.DiscoveryRequest) returns (envoy.api.v2.DiscoveryResponse) {
+    option (google.api.http) = {
+      post: "/v2/discovery:network_policy_ids"
+      body: "*"
+    };
+  }
+}
+
+// The mapping of a host's IP address to its network policy identifier.
+// A host may be associated only one network policy.
+message NetworkPolicyIdentifier {
+  // The unique IP address for the host.
+  // An empty address is not allowed.
+  string host_address = 1 [(validate.rules).string.min_bytes = 1];
```

Comment:
Multiple addresses can have the same policy ID, should make this repeated to reflect that?

Reply:
If multiple hosts have the same policy ID, we'll have one `NetworkPolicyIdentifier` for each host.
The filter will look up by IP address, not by policy ID. So it doesn't make sense to me to group by policy ID.

Reply:
Is the name `NetworkPolicyIdentifier` confusing?

Reply:
The fact that the filter will look by IP does not necessarily mean that NPIDDS message must index by IP. It just needs to carry the information, the recipient can then index on IP. Makes sense?

Reply:
Maybe PolicyIPMapping would be more descriptive?

Reply:
If we did that, then I'd remove the `NetworkPolicyIdentifier` discovery service altogether, and just cram a list of IP addresses into `NetworkPolicy`, since the `NetworkPolicy` discovery service already looks up by network policy ID.

Reply:
Hmmm in retrospect, it'd be nice to keep a separate protocol since the mappings should be resolved once per proxy and shared among all filters in the proxy.

OK, I'll index by network policy ID, and have a list of IP addresses. I'll need to change the name of the DS though.